src/cpp/src/utils.cpp

```diff
@@ -282,6 +283,34 @@ ov::Core singleton_core() {
     return core;
 }
 
+inline bool file_exists(const std::string& name) {
+    std::ifstream f(name.c_str());
+    return f.good();
+}
+
+std::shared_ptr<ov::Model> read_model(const std::filesystem::path& model_dir,  const ov::AnyMap& config) {
+    auto gguf_model_name = model_dir / "openvino_model.gguf";
+
+    if (file_exists(gguf_model_name)) {
```

Comment:
```suggestion
    if (std::filesystem::exists(gguf_model_name)) {
```
no needs in custom functions